libs/libc/syslog/lib_syslog.c

```diff
@@ -52,27 +52,22 @@
 
 void vsyslog(int priority, FAR const IPTR char *fmt, va_list ap)
 {
-  /* Check if this priority is enabled */
-
-  if ((g_syslog_mask & LOG_MASK(LOG_PRI(priority))) != 0)
```

Comment:
why move the filter to kernel space, which impact the performance in kernel/protected build a lot